lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/driver/CommandSender.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.vortex.driver;
+
+import org.apache.commons.lang.SerializationUtils;
+import org.apache.reef.driver.context.ActiveContext;
+import org.apache.reef.vortex.common.Command;
+
+import javax.inject.Inject;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+/**
+ * Takes the serialization work from the scheduler thread.
+ * todo(not urgent): See if we can make changes in REEF internals to
+ * have the RemoteManager Thread pool do this work.
+ */
+public class CommandSender {
```

Comment:
The purpose of the additional wrapper is to allow some sort of protocol between a Driver and the Executors. One potential use is to cancel a (duplicate) tasklet; another is to separate (input) data from code (i.e., VortexFunction) shipping in case multiple tasklets share a same input.